app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/SaveMappingMissionActivityTest.kt

```diff
@@ -124,136 +166,126 @@ class SaveMappingMissionActivityTest {
+
+        verify(mappingMissionDao, times(1)).shareMappingMission(USER_UID, UNNAMED_MISSION)
+        verify(mappingMissionDao, times(1)).storeMappingMission(USER_UID, UNNAMED_MISSION)
     }
 
     @Test
     fun nameIsCorrect() {
         activityRule.scenario.recreate()
 
-        val name = "My mission"
-
-        val expectedMappingMission = MappingMission(name)
-
         onView(withId(R.id.missionName)).perform(click()).perform(
             typeText(
-                name
+                NAMED_MISSION.name
             )
         ).perform(closeSoftKeyboard())
 
         onView(withId(R.id.privateCheckBox)).perform(click())
         onView(withId(R.id.saveButton)).perform(click())
 
-        assertThat(
-            activityRule.scenario.state.toString(),
-            equalTo(Lifecycle.State.DESTROYED.toString())
+        sleep(1000)
+
+        Intents.intended(
+            IntentMatchers.hasComponent(ComponentNameMatchers.hasClassName(ItineraryShowActivity::class.java.name))
         )
 
-        verify(mappingMissionDao, times(1)).storeMappingMission(USER_UID, expectedMappingMission)
-    }
+        val intents = Intents.getIntents()
+        assert(intents.any { it.hasExtra(MissionViewAdapter.FLIGHTHEIGHT_INTENT_PATH) })
+        assert(intents.any { it.hasExtra(MissionViewAdapter.STRATEGY_INTENT_PATH) })
+        assert(intents.any { it.hasExtra(MissionViewAdapter.AREA_INTENT_PATH) })
 
+        verify(mappingMissionDao, times(1)).storeMappingMission(USER_UID, NAMED_MISSION)
+    }
 
     @Test
-    fun checkBoxesEnableSaveButton() {
+    fun saveWithoutCheckingShowDialog() {
         activityRule.scenario.recreate()
 
-        onView(withId(R.id.privateCheckBox))
-            .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
-        onView(withId(R.id.sharedCheckBox))
-            .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
-        onView(withId(R.id.saveButton))
-            .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
-        onView(withId(R.id.saveButton))
-            .check(matches(not(isEnabled())))
-
-
-        // Check private box
-        onView(withId(R.id.privateCheckBox)).perform(click())
-        onView(withId(R.id.saveButton))
-            .check(matches(isEnabled()))
-
-        // Uncheck private box
-        onView(withId(R.id.privateCheckBox)).perform(click())
-        onView(withId(R.id.saveButton))
-            .check(matches(not(isEnabled())))
-
-        // Check share box
-        onView(withId(R.id.sharedCheckBox)).perform(click())
-        onView(withId(R.id.saveButton))
-            .check(matches(isEnabled()))
+        onView(withId(R.id.saveButton)).perform(click())
 
-        // Uncheck share box
-        onView(withId(R.id.sharedCheckBox)).perform(click())
-        onView(withId(R.id.saveButton))
-            .check(matches(not(isEnabled())))
+        onView(withText(R.string.continue_without_saving)).perform(click())
```

Comment:
Same here, I think you should check that it exists
```suggestion
        onView(withText(R.string.continue_without_saving)).check(matches(isDisplayed()))
        onView(withText(R.string.continue_without_saving)).perform(click())
```